### PR TITLE
Issue #2942871 by Makishima: Add reCAPTCHA to sign up form

### PR DESCRIPTION
**Pull request by VolodymyrMartsynenko**
## Problem
Add captcha (reCAPTCHA) to the sign up form

## Solution

- Add reCAPTCHA module to composer.json
- Create Social CAPTCHA (social_captcha) module and add it to social_features
- Add hook_update to social_core module which enable social_captcha module

## Issue tracker
- https://www.drupal.org/project/social/issues/2942871

## HTT
- [x] Check out the code changes
- [x] Go to Sign up form (/user/register)
- [x] You should see captcha in the bottom of the form  
- [x] As admin you should configure reCaptha module, add Google reCaptcha Site key and  Secret key on the configuration page (/admin/config/people/captcha/recaptcha)
- [x] Log out. As anonymous user you should see reCaptcha in the bottom of the Sign up form

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -117,6 +117,7 @@
         "bower-asset/highlight": "9.7.0",
         "embed/embed": "2.7.7",
         "npm-asset/jquery.caret": "^0.3.1",
-        "npm-asset/diff": "^3.4"
+        "npm-asset/diff": "^3.4",
+        "drupal/recaptcha": "2.2"
```
Comment (bramtenhove) [inline composer.json]: Maybe we can move `drupal/recaptcha` more to the top where also the other drupal dependencies are

Comment (bramtenhove) [reply composer.json]: Also we need to add this to `.make` file

Comment (jaapjan) [reply composer.json]: Done :-)

**Comment by bramtenhove** [comment]
@VolodymyrMartsynenko I noticed there is a label "CAPTCHA" above the Captcha. This looks off, can you remove it?

**Comment by bramtenhove** [comment]
@VolodymyrMartsynenko could you take a look at my comment?

**Pull request closed without merging**